misc/package/Data Files/MWSE/core/mcm/components/templates/Template.lua

```diff
@@ -21,9 +23,16 @@ function Template:new(data)
 	local pages = {}
 	t.pages = t.pages or {}
 	for _, page in ipairs(t.pages) do
-		page.class = page.class or "Page"
-		local newPage = self:getComponent(page)
-		table.insert(pages, newPage)
+		-- Make sure it's actually a `Page`.
+		if not page.componentType then
+			local componentClass = fileUtils.getComponentClass(page.class or "Page")
+			if not componentClass then
+				error(string.format("Could not intialize page %q", page.label))
```

Comment:
Again, let's not throw an error here, but log a warning.

Reply:
In this case, the `error` I added would be inconsistent with the previous implementation since `table.insert(pages, nil)` wouldn't do anything. 

It seems a bit inconsistent to me that in the previous implementation, `getComponent` resulted in an error in the `Category` code, but not in the `Template` code.

I'm leaning towards changing both the `Category` and `Template` code so they issue a warning and ignore the faulty components instead of throwing an error, but wanted to know what you think before I made further changes.

Do you think I should move ahead and make those changes?

Reply:
In other words, based on your comment on my concern from above, this change shouldn't break any existing mod's MCM, since it would still error out, but at at different point, and possibly with a bit worse error message. I think we are good as is.